src/bbs2gh/Handlers/MigrateRepoCommandHandler.cs

```diff
@@ -437,10 +442,7 @@ private void ValidateOptions(MigrateRepoCommandArgs args)
                 }
             }
 
-            if (ShouldDownloadArchive(args))
-            {
-                ValidateDownloadOptions(args);
-            }
+            ValidateDownloadOptions(args);
```

Comment:
We can actually run this validation method even if we're not downloading - the method is safe to run. This allows us to include the new `--archive-download-host can only be provided if SSH or SMB download options are provided.` validation inside the method too.